### PR TITLE
`ManagedIdentityCredential` honors CancellationTokens

### LINKED ISSUE #47156: [BUG] ManagedIdentity.GetTokenAsync method does not handle cancellationtoken
### Library name and version

Azure.Identity v1.13.x

### Describe the bug

Since v1.13.0 release the GetTokenAsync method on ManagerdIdentity does not respect/handle cancellationToken and will only timeout after 100 seconds.   

This worked just fine in older version (up to v1.12.1)

### Expected behavior

The GetTokenAsync method should return a TaskCanceledException after the cancellationToken has requested cancellation as it did in v1.12.1 and older.

### Actual behavior

When running with v1.13.x this will run for 100 seconds and return this message: 

ManagedIdentityCredential authentication failed: Retry failed after 5 tries. Retry settings can be adjusted in ClientOptions.Retry or by configuring a custom retry policy in ClientOptions.RetryPolicy. 
(A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond. (169.254.169.254:80)) 
(A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond. (169.254.169.254:80)) 
(A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond. (169.254.169.254:80)) 
(A connection attempt failed because the connected party did not properly respond after a period of time, or established connection failed because connected host has failed to respond. (169.254.169.254:80)) (The operation was canceled.)
See the troubleshooting guide for more information. https://aka.ms/azsdk/net/identity/managedidentitycredential/troubleshoot

MsalServiceException•••Retry failed after 5 tries. Retry settings can be adjusted in ClientOptions.Retry or by configuring a custom retry policy in ClientOptions.RetryPolicy... | MsalServiceException••• | 
Retry failed after 5 tries. Retry settings can be adjusted in ClientOptions.Retry or by configuring a custom retry policy in ClientOptions.RetryPolicy...
MsalServiceException
Retry failed after 5 tries. Retry settings can be adjusted in ClientOptions.Retry or by configuring a custom retry policy in ClientOptions.RetryPolicy..._

### Reproduction Steps

````
var credential = new ManagedIdentityCredential();
var scopes = "api://52940e13-4781-4ef7-a380-01494c61b4b7";

var tokenRequestContext = new TokenRequestContext(scopes.Split(new []{','}, StringSplitOptions.RemoveEmptyEntries));

// so we will drop this down to 5 seconds 
var myCts = new CancellationTokenSource();
myCts.CancelAfter(5000);  // 5000ms = 5s
var token = await credential.GetTokenAsync(tokenRequestContext, myCts.Token);

````

### Environment

.NET SDK:
 Version:           9.0.100
 Commit:            59db016f11
 Workload version:  9.0.100-manifests.c6f19616
 MSBuild version:   17.12.7+5b8665660

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22631
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\9.0.100\

### PR DESCRIPTION
**Pull request by christothes**
fixes #47156

### CODE REVIEW

File: sdk/core/Azure.Core/CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 ### Bugs Fixed
 
 - Fixed an issue where `BearerTokenAuthenticationPolicy` throws `ArgumentOutOfRangeException` if the `ExpiresOn` property of the token is the default value. ([#47040](https://github.com/Azure/azure-sdk-for-net/pull/47040))
+- Fixed an issue where `ManagedIdentityCredential` does not honor the `CancellationToken` passed to `GetToken` and `GetTokenAsync`. ([#47156](https://github.com/Azure/azure-sdk-for-net/issues/47156))
```
Comment (annelo-msft) [inline sdk/core/Azure.Core/CHANGELOG.md:12]: Curious: if the fix is in the Identity library, why add the comment to the Azure.Core changelog?

Comment (christothes) [reply sdk/core/Azure.Core/CHANGELOG.md:12]: That would be because I had the wrong editor window open at the time 🤦‍♂️

Comment (annelo-msft) [reply sdk/core/Azure.Core/CHANGELOG.md]: You would not be the first person to have done such a thing! :)

**Review by jsquire — APPROVED** [review]

**Review by chlowell — APPROVED** [review]

**Comment by christothes** [comment]
@microsoft-github-policy-service rerun

**Pull request merged**